### PR TITLE
chore: Use isAllowedOrigin in AddonReviewList

### LINKED ISSUE #2528: addonReviewList not using isAllowedOrigin
EDIT: Issue updated; I should just change my CDN host but we should fix the `AddonReviewList` component to check origin. I noticed this issue in the first place because it didn't check.

------

Original issue:

If you use `API_HOST=https://addons.mozilla.org yarn dev:amo` as I often do for more robust data for testing, you'll notice all the add-on icons are missing. Actually, a lot of data is missing for any icon that uses `isAllowedOrigin`. I'm guessing we do this so we only even bother rendering images for CDN hosts we allow and otherwise use the fallback icon which is nice, but I'd like an override for development so I can view all the remote content when I'm working with external server data.

Interestingly the `AddonReviewList` component doesn't use it so it shows the icon.

### PR DESCRIPTION
**Pull request by tofumatt**
Fixes #2528.

Just adds this check to the AddonReviewList.

### CODE REVIEW

**Comment by tofumatt** [comment]
Cool, this is ready now 😄

File: src/core/utils.js [hunk 1]
```diff
@@ -123,14 +123,18 @@ export function loadAddonIfNeeded(
   return _refreshAddon({ addonSlug: slug, apiState: state.api, dispatch });
 }
 
-export function isAllowedOrigin(urlString, { allowedOrigins = [config.get('amoCDN')] } = {}) {
+export function isAllowedOrigin(urlString, {
+  allowedOrigins = [config.get('amoCDN')],
+  allOriginsAllowed = config.get('allOriginsAllowed'),
```
Comment (muffinresearch) [inline src/core/utils.js]: Where is this used in the function?

Comment (muffinresearch) [inline src/core/utils.js]: Ah I see I think this can be removed now.

Comment (tofumatt) [inline src/core/utils.js]: Oops, yeah sorry should have removed it. Probably why the tests are failing too (lint). Will send a patch in a sec.

File: tests/unit/amo/components/TestAddonReviewList.js [hunk 2]
```diff
@@ -114,12 +115,23 @@ describe('amo/components/AddonReviewList', () => {
       expect(byLine).toContain(fakeReview.user.name);
     });
 
-    it('renders an icon in the header', () => {
+    it('renders the fallback icon if the origin is not allowed', () => {
```
Comment (kumar303) [inline tests/unit/amo/components/TestAddonReviewList.js]: The descriptions of these two tests look like they're swapped, no? This test looks like it expects a real icon URL and the other expects a fallback.

Comment (tofumatt) [inline tests/unit/amo/components/TestAddonReviewList.js]: Oops, you are correct. My bad 😞

**Pull request merged**